### PR TITLE
Draw a state-maintained snake in the game

### PR DESCRIPTION
**Pull request by sashaweiss**
- Add snake state to `Game`
- Fixes to updating snake state
- Render snake state in `Terminal`

### CODE REVIEW

**Comment by sashaweiss** [comment]
@nathanshelly went through and cleanup up some, lmk what you think (granted we wrote this together but whatever)

File: src/snek/game.rs [hunk 1]
```diff
@@ -37,23 +53,63 @@ impl Game {
...
+    snake.grow(SnakeDirection::North);
+    snake.grow(SnakeDirection::North);
+    snake.grow(SnakeDirection::East);
+    snake.grow(SnakeDirection::East);
+    snake.grow(SnakeDirection::East);
+
     Game {
       dimensions,
-      snek: Snake::new(),
+      snek: snake,
```
Comment (nathanshelly) [inline src/snek/game.rs:67]: lol at some point we should have the constructor generate a snake with one segment starting in the center but could be future pr

File: src/snek/mod.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #![allow(clippy::trivially_copy_pass_by_ref)]
+#![allow(dead_code)] // TODO: @sashaweiss @nathanshelly remove this once MVP is working
```
Comment (nathanshelly) [inline src/snek/mod.rs:2]: ❤️ this `TODO`

File: src/snek/terminal.rs [hunk 3]
```diff
@@ -106,3 +167,12 @@ impl GameDimensions {
...
 }
+
+impl GameCoordinate {
+  fn adjusted(self, TerminalOffset { x, y }: TerminalOffset) -> Self {
+    GameCoordinate {
+      x: self.x + x,
+      y: self.y + y,
+    }
+  }
+}
```
Comment (nathanshelly) [inline src/snek/terminal.rs]: don't think we need this since we ended up defining the `adjust` closure

**Review by nathanshelly — APPROVED** [review]
🐍

File: src/snek/game.rs (hunk 1 above)
Comment (sashaweiss) [reply src/snek/game.rs:67]: yeah i figure future PR is alright for this - it like helps the demo make sense if nothing else

File: src/snek/terminal.rs (hunk 3 above)
Comment (sashaweiss) [reply src/snek/terminal.rs]: good call

Comment (sashaweiss) [reply src/snek/terminal.rs]: ironically, this (and other comments) are slipping by because of that allow(dead_code) lol

**Comment by sashaweiss** [comment]
interesting - it seems like because previous checks had passed I was able to merge before CI had finished when I pushed that last update?

**Comment by nathanshelly** [comment]
> interesting - it seems like because previous checks had passed I was able to merge before CI had finished when I pushed that last update?

huh not sure why that would be but everything seems fine so 🤷‍♂

**Pull request merged**